robolectric-shadows/shadows-core/src/main/resources/org/robolectric/shadows/ShadowViewRootImpl.java.vm

```diff
@@ -0,0 +1,90 @@
+package org.robolectric.shadows;
+
+import static org.robolectric.internal.Shadow.directlyOn;
+
+import android.content.res.Configuration;
+import android.graphics.Rect;
+import android.os.Looper;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewRootImpl;
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+import org.robolectric.annotation.RealObject;
+import org.robolectric.util.ReflectionHelpers.ClassParameter;
+
+/**
+ * Shadow for {@link android.view.ViewRootImpl}.
+ */
+@Implements(value = ViewRootImpl.class, isInAndroidSdk = false)
+public class ShadowViewRootImpl {
+  public static final String VIEW_ROOT_IMPL_CLASS_NAME = "android.view.ViewRootImpl";
+
+  @RealObject
+  Object realObject;
+
+  @Implementation
+  public static Object getWindowSession(Looper mainLooper) {
+    return null;
+  }
+
+  @Implementation
+  public void playSoundEffect(int effectId) {}
+
+#if ($api >= 21)
+  @Implementation
+  public void dispatchResized(Rect frame, Rect overscanInsets, Rect contentInsets,
+      Rect visibleInsets, Rect stableInsets, boolean reportDraw, Configuration newConfig) {
+    directlyOn(realObject, VIEW_ROOT_IMPL_CLASS_NAME, "dispatchResized",
```

Comment:
I followed the pattern in ShadowWindowManagerImpl.java

  @Implementation
  public void removeView(View view) {
    views.remove(view);
    directlyOn(realObject, WINDOW_MANAGER_IMPL_CLASS_NAME, "removeView", ClassParameter.from(View.class, view));
  }